kexdh.c

```diff
@@ -200,4 +302,48 @@ kex_dh_dec(struct kex *kex, const struct sshbuf *dh_blob,
 	sshbuf_free(buf);
 	return r;
 }
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
```

Comment:
Again, this is bringing in OpenSSL 3.0 stuff, not just converting to EVP_PKEY. I think a first step here is to convert to PKEY *without* including anything for 3.0, and to just have a minimal change that converts to the EVP_PKEY api with the 1.1.1. interface.